x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/analytics/AnalyticsCollection.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.entsearch.analytics;
+
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.xcontent.ObjectParser;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xcontent.XContentParserConfiguration;
+import org.elasticsearch.xcontent.XContentType;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * The {@link AnalyticsCollection} model.
+ */
+public class AnalyticsCollection implements Writeable, ToXContentObject {
```

Comment:
Is this really needed? Are we going to add more than just the name of the collection here?

Reply:
My assumption is that the AnalyticsCollection model will requires more than a data stream in the near future (e.g when the user model will be added to it).I think that making sure that the model implements the serialization stuffs from the very beginning will make these persistence changes easier to implement.

Reply:
Fair, thanks